src/test/java/org/kohsuke/github/AppTest.java

```diff
@@ -1608,8 +1609,8 @@ public void testRepoLabel() throws IOException {
             try {
                 t = r.getLabel("test");
                 fail("Test label should be deleted.");
-            } catch (IOException ex) {
-                assertThat(ex, instanceOf(FileNotFoundException.class));
+            } catch (UncheckedIOException ex) {
+                assertThat(ex.getCause(), instanceOf(FileNotFoundException.class));
```

Comment:
This going to be a massive issue.  

Consumers of this library that expect to be able to detect missing items by catching `FileNotFoundException` will have to catch `UncheckedIOException` and then check the type of `e.getCause()`.  That's not so bad per se, but it is a significant change in behavior.    

We could handled this be replacing the generic `UncheckedIOException` with an `UncheckedFileNotFoundException`, but it would still be a significant change not something that could be used by existing consumers without updates.

Reply:
Is it just one more thing people must consider and adjust when switching from 1.x to 2.x? As a maintainer, I would put a bold notice about all these things in a changelog.